x/provider/ante/ante_test.go

```diff
@@ -0,0 +1,151 @@
+package ante
+
+import (
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/crypto/ed25519"
+	"github.com/tendermint/tendermint/libs/log"
+	dbm "github.com/tendermint/tm-db"
+
+	"github.com/ovrclk/akash/x/provider/keeper"
+	"github.com/ovrclk/akash/x/provider/types"
+)
+
+var (
+	key1pub  = ed25519.GenPrivKey().PubKey()
+	key1Addr = sdk.AccAddress(key1pub.Address())
+	key2pub  = ed25519.GenPrivKey().PubKey()
+	key2Addr = sdk.AccAddress(key2pub.Address())
+)
+
+type TestSuite struct {
+	suite.Suite
+	ms     sdk.CommitMultiStore
+	ctx    sdk.Context
+	keeper keeper.Keeper
+	chain  sdk.AnteHandler
+}
+
+func createTestCodec() *codec.Codec {
+	cdc := codec.New()
+	auth.RegisterCodec(cdc)
+	bank.RegisterCodec(cdc)
+	sdk.RegisterCodec(cdc)
+	types.RegisterCodec(cdc)
+	codec.RegisterCrypto(cdc)
+
+	return cdc
+}
+
+func TestTestSuite(t *testing.T) {
+	suite.Run(t, new(TestSuite))
+}
+
+func (s *TestSuite) SetupSuite() {
+	keyProvider := sdk.NewTransientStoreKey(types.StoreKey)
+
+	db := dbm.NewMemDB()
+	s.ms = store.NewCommitMultiStore(db)
+	s.ms.MountStoreWithDB(keyProvider, sdk.StoreTypeIAVL, db)
+
+	err := s.ms.LoadLatestVersion()
+	require.NoError(s.T(), err)
+
+	s.ctx = sdk.NewContext(s.ms, abci.Header{}, true, log.NewNopLogger())
+
+	s.keeper = keeper.NewKeeper(createTestCodec(), keyProvider)
+
+	s.chain = sdk.ChainAnteDecorators(
+		NewDecorator(s.keeper),
+	)
```

Comment:
At some point you should test that `mergeAnteHandlers()` function (in `./app` I believe?)